picard/script/functions.py

```diff
@@ -760,14 +760,22 @@ def func_lenmulti(parser, multi, separator=MULTI_VALUED_JOINER):
     """`$performer(pattern="",join=", ")`
 
 Returns the performers where the performance type (e.g. "vocal") matches `pattern`, joined by `join`.
+`pattern` can be a regular expression.
 
 _Since Picard 0.10_"""
 ))
 def func_performer(parser, pattern="", join=", "):
     values = []
     for name, value in parser.context.items():
-        if name.startswith("performer:") and pattern in name:
-            values.append(value)
+        if name.startswith("performer:"):
+            name, performance = name.split(':', 2)
+            try:
+                match = bool(re.search(pattern, performance))
```

Comment:
I wonder about case sensitity here. Is `name` guaranteed to be lowercased? Should we do a case-insensitive match?
Do we expect `$performer(^GUITAR\$)` to match?

Reply:
Yes, I think we should compare it case insensitive. It's definitely not guaranteed to be lowercase. It usually probably is, as it is usually filled by MB data, but even there is no full guarantee that instruments are allways lowercase.

Reply:
We should convert whole `name` to lowercase and use case-insensitive regex matching (`re.search` flag) and lowercase `pattern` if the fallback section.

Reply:
Given that the regex is going to be run on every track, should we cache the compiled regex?